src/renderer/src/routes/(MapTabs)/_Map.test.tsx

```diff
@@ -0,0 +1,20 @@
+import { render, screen } from '@testing-library/react'
+import { expect, test, vi } from 'vitest'
+
+import { MapLayout } from './_Map'
+
+vi.mock('@tanstack/react-router', () => ({
+	useNavigate: vi.fn(() => {
+		return { navigate: vi.fn() }
+	}),
+	createFileRoute: vi.fn(() => {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		return (options: any) => ({ component: options.component }) // Mocked implementation
+	}),
+	Outlet: () => <div>Mocked Outlet</div>,
+}))
```

Comment:
just for my understanding, this has to be done for each test that relies on the router in some way?

Reply:
it depends. I think short answer yes. long answer, if we want to test actual navigation actions, we need to stub out the router instead of just stubbing out the function inside the router. 

I think more work needs to go into creating our mocks and stubs, but this just allows it run (its was basically a smoke screen test for jsdom)